src/components/Table/GridTable.tsx

```diff
@@ -1474,24 +1474,25 @@ function filterRows<R extends Kinded>(
   filter: string | undefined,
 ): ParentChildrenTuple<R>[] {
   // Make a function to do recursion
-  function filterFn(acc: ParentChildrenTuple<R>[], row: GridDataRow<R>): ParentChildrenTuple<R>[] {
+  function filterFn(acc: ParentChildrenTuple<R>[], row: GridDataRow<R>, parentMatches?: boolean): ParentChildrenTuple<R>[] {
     // Break up "foo bar" into `[foo, bar]` and a row must match both `foo` and `bar`
     const filters = (filter && filter.split(/ +/)) || [];
     const matches =
       row.kind === "header" ||
       row.kind === "totals" ||
       filters.length === 0 ||
       !!row.pin ||
+      (filters.length > 0 && parentMatches) || //matching childrens if parent matches to display the nested records on searches
```

Comment:
Question: Can this row be just `parentMatches ||`, i.e. do we need the `filters.length > 0` check?